java/vector/src/main/codegen/templates/ArrowType.java

```diff
@@ -246,6 +251,61 @@ public boolean equals(Object obj) {
   }
   </#list>
 
+  /**
+   * A user-defined data type that wraps an underlying storage type.
+   */
+  public abstract static class ExtensionType extends ComplexType {
+    /** The on-wire type for this user-defined type. */
+    public abstract ArrowType storageType();
+    /** The name of this user-defined type. Used to identify the type during serialization. */
+    public abstract String extensionName();
+    /** Check equality of this type to another user-defined type. */
+    public abstract boolean extensionEquals(ExtensionType other);
+    /** Save any metadata for this type. */
+    public abstract String serialize();
+    /** Given saved metadata and the underlying storage type, construct a new instance of the user type. */
+    public abstract ArrowType deserialize(ArrowType storageType, String serializedData);
+    /** Construct a vector for the user type. */
+    public abstract FieldVector getNewVector(String name, FieldType fieldType, BufferAllocator allocator);
+
+    /** The field metadata key storing the name of the extension type. */
+    public static final String METADATA_KEY_NAME = "arrow_extension_name";
```

Comment:
i think this is what you mentioned was out of date?

Reply:
Yes, the constants need updating...